File: SpringBoot/src/main/java/com/hospital/qlbv/entity/MedicalRecord.java
package com.hospital.qlbv.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "medical_records")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class MedicalRecord {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer recordId;
    
    @Column(name = "patient_id")
    private Integer patientId;
    
    @Column(name = "doctor_id")
    private Integer doctorId;
    
    @Column(name = "appointment_id")
    private Integer appointmentId;
    
    private String symptoms;
    private String diagnosis;
    private String treatment;
    private String notes;
    private String status;
    
    @Column(name = "date_created")
    private String dateCreated;
    
    @Column(name = "updated_at")
    private String updatedAt;
    
    @Column(name = "vital_signs")
    private String vitalSigns;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "patient_id", insertable = false, updatable = false)
    @JsonIgnoreProperties({"user", "insurance"})
    private Patient patient;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "doctor_id", insertable = false, updatable = false)
    @JsonIgnoreProperties({"user", "department"})
    private Doctor doctor;
    
    @OneToMany(mappedBy = "medicalRecord", cascade = CascadeType.ALL)
    @JsonIgnoreProperties({"medicalRecord"})
    @Builder.Default
    private List<Prescription> prescriptions = new ArrayList<>();
    
    @OneToMany(mappedBy = "medicalRecord", cascade = CascadeType.ALL)
    @JsonIgnoreProperties({"medicalRecord"})
    @Builder.Default
    private List<TestRequest> testRequests = new ArrayList<>();
}

File: Backend/Models/MedicalRecord.cs
using System.ComponentModel.DataAnnotations;

namespace QuanLyBenhVien.API.Models;

public class MedicalRecord
{
    [Key]
    public int RecordID { get; set; }
    public int? PatientID { get; set; }
    public int? DoctorID { get; set; }
    public int? AppointmentID { get; set; }
    public string? Symptoms { get; set; }
    public string? Diagnosis { get; set; }
    public string? Treatment { get; set; }
    public string? Notes { get; set; }
    public string? Status { get; set; }
    public DateTime? DateCreated { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public string? VitalSigns { get; set; }

    // Navigation properties
    public virtual Patient? Patient { get; set; }
    public virtual Doctor? Doctor { get; set; }
    public virtual Appointment? Appointment { get; set; }
    public virtual ICollection<Prescription> Prescriptions { get; set; } = new List<Prescription>();
    public virtual ICollection<TestRequest> TestRequests { get; set; } = new List<TestRequest>();
}
